fastlane/lanes/build.rb

```diff
@@ -266,7 +275,9 @@
     )
   end
 
-  lane :build_for_app_store_connect_reader do
+  lane :build_for_app_store_connect_reader do |build_number: ENV.fetch('BUILDKITE_BUILD_NUMBER', nil)|
+    UI.user_error!('No build number provided and BUILDKITE_BUILD_NUMBER environment variable is not set') if build_number.nil?
```

Comment:
Either before merging or in a followup, I think it would be better to remove the default `BUILDKITE_BUILD_NUMBER` value in favor of having the CI script passing it.

This way, we remove one bit of coupling between Buildkite and this lane.